Sources/Sentry/include/HybridPublic/SentryEnvelopeItemType.h

```diff
@@ -3,6 +3,7 @@
 static NSString *const SentryEnvelopeItemTypeEvent = @"event";
 static NSString *const SentryEnvelopeItemTypeSession = @"session";
 static NSString *const SentryEnvelopeItemTypeUserFeedback = @"user_report";
+static NSString *const SentryEnvelopeItemTypeFeedback = @"feedback";
```

Comment:
`h`: You most likely also have to add a `SentryDataCategory` and an entry to `sentryDataCategoryForEnvelopItemType` for client reports to work correctly.

Reply:
Thanks for pointing it out 👍🏻 I'm not intimately familiar with all this machinery.

Reply:
That's what I'm here for 😅

Reply:
A question, while we're on the subject: what is the actual difference between e.g. `SentryEnvelopeItemTypeUserFeedback` and `kSentryDataCategoryNameUserFeedback`? It looks like we pass `SentryEnvelopeItemTypeUserFeedback` into `sentryDataCategoryForEnvelopItemType`  in at least one place: https://github.com/getsentry/sentry-cocoa/blob/37fd7a27e43b84eddf942bce1a0066d5cbc050a0/Sources/Sentry/SentryEnvelopeRateLimit.m#L57-L58 which then later spits out `kSentryDataCategoryNameUserFeedback` in the call to `nameForSentryDataCategory` here: https://github.com/getsentry/sentry-cocoa/blob/37fd7a27e43b84eddf942bce1a0066d5cbc050a0/Sources/Sentry/SentryHttpTransport.m#L167. Couldn't we deduplicate the string defs?

Reply:
No sadly not. Envelope item types and data categories look similar, but they have slight differences. I'm going to open a PR to add a comment. This also confused me quite a bit in the past.
![CleanShot 2024-11-20 at 16 58 29@2x](https://github.com/user-attachments/assets/9c89308f-ba7b-4c17-b873-d755d6198906)

Reply:
PR to clarify that: https://github.com/getsentry/sentry-cocoa/pull/4556

Reply:
Thanks, I tried to look for the differences but I guess I was only comparing members that were common between the two sets and missed the ones that only existed in one or the other!